resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala

```diff
@@ -48,8 +49,8 @@ class MockResolver extends SparkRackResolver {
 class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfterEach {
   val conf = new YarnConfiguration()
   val sparkConf = new SparkConf()
-  sparkConf.set("spark.driver.host", "localhost")
-  sparkConf.set("spark.driver.port", "4040")
+  sparkConf.set(DRIVER_HOST_ADDRESS.key, "localhost")
+  sparkConf.set(DRIVER_PORT.key, "4040")
```

Comment:
`SparkConf#set` is an overloaded method if you pass a `ConfigEntry` as a first argument it will be typesafe (in this case for `DRIVER_PORT` an Int is expected as second argument).

If the first arg of `SparkConf#set` is a String then the value must be passed as String (not typesafe).

I would prefer to use the typesafe setter and for this only the `.key` method calls should be removed from the ConfigEntries you are using at the calls of `SparkConf#set`.

For the launcher (like in BaseYarnClusterSuite.scala) you can only pass String key when `setConf` is called.

Reply:
Thank you for your suggestion. I will fix it。